hass-addon-sunsynk-dev/run.py

```diff
@@ -71,29 +71,49 @@ def _handler(value):
     for filt in SENSORS:
         sensor = filt.sensor
 
+        entity_name = f"{OPT.sensor_prefix} {sensor.name}".strip()
```

Comment:
There seems to be lots of duplication here.
Can you extract all the common attributes? 

```python
ent = {
entity_name: f"{OPT.sensor_prefix} {sensor.name}".strip(),
device=dev
}
```

Then add specific values in the various if statements and call it with the options, i.e.:

```python
NumberEntity(**ent)
NumberEntity(**ent)
SelectRWSensor(**ent)
```

You can even delete keys from options if not required for that sensor (preferably dont use this too often)

Reply:
Excellent, I wondered if there was a better way to do this and fully agree. Python is not my first language as you can tell :) Will update tonight.

Reply:
It works but one has to quote the dictionary keys to avoid a `Pylint(E0602:undefined-variable)` error.

<img width="528" alt="image" src="https://user-images.githubusercontent.com/16413523/183091559-56c0f249-3ab4-49e5-8747-4a356d92f818.png">
<img width="527" alt="image" src="https://user-images.githubusercontent.com/16413523/183091634-3b2f0047-7aac-4043-b080-b9a93e59dadc.png">

Here is the full version with quoted dictionary keys. All duplication has been removed, which is great. The only minor issue is that fields in `ent` are quoted whilst entity-specific fields are not since they are passed in as named arguments, but it seems to make no difference with validation - having an invalid dictionary key in `ent` or an invalid named argument when constructing the specific entity yields the same TypeError at runtime.

```python
ent = {
    "device": dev,
    "name": f"{OPT.sensor_prefix} {sensor.name}".strip(),
    "state_topic": f"{SS_TOPIC}/{OPT.sunsynk_id}/{sensor.id}",
    "unique_id": f"{OPT.sunsynk_id}_{sensor.id}",
    "unit_of_measurement": sensor.unit,
}
if isinstance(sensor, RWSensor):
    ent = {
        **ent,
        "command_topic": f"{SS_TOPIC}/{OPT.sunsynk_id}/{sensor.id}_set",
        "entity_category": "config",
    }

if isinstance(sensor, NumberRWSensor):
    ents.append(
        NumberEntity(
            **ent,
            min=float(sensor.min_value),
            max=float(sensor.max_value),
            on_change=create_on_change_handler(filt, int),
        )
    )
    continue

if isinstance(sensor, SelectRWSensor):
    ents.append(
        SelectEntity(
            **ent,
            options=sensor.available_values(),
            on_change=create_on_change_handler(filt, str),
        )
    )
    continue

ents.append(SensorEntity(**ent))
```